Source/Services/Beam/beam_manager_impl.cpp

```diff
@@ -939,43 +952,19 @@ beam_manager_impl::on_socket_message_received(
     _In_ const string_t& message
 )
 {
-    try
-    {
-        web::json::value msgJson = web::json::value::parse(message);
-
-        if (msgJson.has_field(RPC_TYPE))
-        {
-            string_t messageType = msgJson[RPC_TYPE].as_string();
-
-            if (0 == messageType.compare(RPC_REPLY))
-            {
-                process_reply(msgJson);
-            }
-            else if (0 == messageType.compare(RPC_METHOD))
-            {
-                process_method(msgJson);
-            }
-            else
-            {
-                LOGS_INFO << "Unexpected message from service";
-            }
-        }
-    }
-    catch (std::exception e)
-    {
-        LOGS_ERROR << "Failed to parse incoming socket message";
-    }
+    std::shared_ptr<beam_rpc_message> rpcMessage = std::shared_ptr<beam_rpc_message>(new beam_rpc_message(get_next_message_id(), web::json::value::parse(message), unix_timestamp_in_ms()));
```

Comment:
doesn't work for private ctors - but you are correct 

---
In reply to: [115872431](https://github.com/WatchBeam/interactive-sdk-cpp/pull/5#discussion_r115872431) [](ancestors = 115872431)